src/toil/test/lib/dockerTest.py

```diff
@@ -1,226 +1,309 @@
 import logging
 import signal
 import time
+import os
 import uuid
+import docker
 from threading import Thread
-from subprocess import CalledProcessError
+from docker.errors import ContainerError
 
-import os
-from pwd import getpwuid
 from bd2k.util.files import mkdir_p
 from toil.job import Job
 from toil.leader import FailedJobsException
-from toil.lib.docker import dockerCall, dockerCheckOutput, _containerIsRunning, _dockerKill, STOP, FORGO, RM
-from toil.test import ToilTest, slow
+from toil.test import ToilTest
+from toil.lib import FORGO, STOP, RM
+from toil.lib.docker import dockerCall, dockerCheckOutput, \
+                            _containerIsRunning, _dockerKill
 
-_log = logging.getLogger(__name__)
+_logger = logging.getLogger(__name__)
 
 
 class DockerTest(ToilTest):
     """
     Tests dockerCall and ensures no containers are left around.
 
-
-    When running tests you may optionally set the TOIL_TEST_TEMP environment variable to the path
-    of a directory where you want temporary test files be placed. The directory will be created
-    if it doesn't exist. The path may be relative in which case it will be assumed to be relative
-    to the project root. If TOIL_TEST_TEMP is not defined, temporary files and directories will
-    be created in the system's default location for such files and any temporary files or
-    directories left over from tests will be removed automatically removed during tear down.
+    When running tests you may optionally set the TOIL_TEST_TEMP environment
+    variable to the path of a directory where you want temporary test files be
+    placed. The directory will be created if it doesn't exist. The path may be
+    relative in which case it will be assumed to be relative to the project
+    root. If TOIL_TEST_TEMP is not defined, temporary files and directories will
+    be created in the system's default location for such files and any temporary
+    files or directories left over from tests will be removed automatically
+    removed during tear down.
     Otherwise, left-over files will not be removed.
     """
     def setUp(self):
         self.tempDir = self._createTempDir(purpose='tempDir')
+        self.dockerTestLogLevel = 'INFO'
 
-    @slow
-    def testDockerClean(self, caching=True):
+    def testDockerClean(self,
+                        disableCaching=True,
+                        detached=True,
+                        rm=True,
+                        deferParam=None):
         """
-        Run the test container that creates a file in the work dir, and sleeps for 5 minutes.  Ensure
-        that the calling job gets SIGKILLed after a minute, leaving behind the spooky/ghost/zombie
-        container. Ensure that the container is killed on batch system shutdown (through the defer
-        mechanism).
-        This inherently also tests _docker
-        :returns: None
+        Run the test container that creates a file in the work dir, and sleeps
+        for 5 minutes.
```

Comment:
5 minutes seems like a long time to wait.

Reply:
The "spooky test container" was not created by me, but in actuality, each is created and killed in about ~10 seconds.  The five minute thing is, I believe, just a secondary backup to making sure it implodes if the tests fail and one doesn't want to manually hunt them down.

Reply:
Ah gotcha, thanks for the explanation.